### PR TITLE
UIIN-2433  Inventory: Retain Search/Browse query/options/filter and facet selections UNLESS user resets/clear selections

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Retain Search/Browse query/options/filter and facet selections UNLESS user resets/clear selections
Store last search mode in `LastSearchTermsProvider` context

## Screenshots

https://github.com/folio-org/ui-inventory/assets/19309423/af3ebb01-7ef4-49d5-834a-a5f4da7774ae

## Issues
[UIIN-2433](https://issues.folio.org/browse/UIIN-2433)

## Related PRs
https://github.com/folio-org/stripes-smart-components/pull/1364

### CODE REVIEW

**Comment by BogdanDenis** [comment]
@folio-org/fe-tl-reviewers please take a look

File: src/components/FilterNavigation/FilterNavigation.js [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
+  const { getLastSearch } = useLastSearchTerms();
+
+  return (
+    <ButtonGroup
+      fullWidth
+      data-test-filters-navigation
+    >
+      {
+        Object.keys(segments).map((name) => {
+          const sort = segmentsSortBy.find(x => x.name === name).sort || '';
```
Comment (Terala-Priyanka) [inline src/components/FilterNavigation/FilterNavigation.js:24]: Looks like this declaration is no more used. Could you reevaluate the need?

File: src/components/InstancesList/InstancesList.js [hunk 2]
```diff
@@ -123,6 +123,7 @@ class InstancesList extends React.Component {
     location: PropTypes.shape({
       search: PropTypes.string,
       state: PropTypes.object,
+      pathname: PropTypes.string,
```
Comment (Terala-Priyanka) [inline src/components/InstancesList/InstancesList.js:126]: Could you check if it is possible to further increase the coverage on this file?

**Review by Terala-Priyanka — COMMENTED** [review]
Looks like there are some probably linting errors listed in "Unchanged files with check annotations". If possible could you fix them?

File: src/components/FilterNavigation/FilterNavigation.js [hunk 3]
```diff
@@ -1,43 +1,54 @@
...
+              >
+                <FormattedMessage id={`ui-inventory.filters.${name}`} />
+              </Button>
+            </>
+          );
+        })
+      }
+    </ButtonGroup>
+  );
+}
```
Comment (Terala-Priyanka) [inline src/components/FilterNavigation/FilterNavigation.js:48]: Could you fix linting error here by adding semi colon?

**Review by Terala-Priyanka — APPROVED** [review]

File: src/components/SearchModeNavigation/SearchModeNavigation.test.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
-import { screen } from '@folio/jest-config-stripes/testing-library/react';
+import { fireEvent, screen } from '@folio/jest-config-stripes/testing-library/react';
 import {
   MemoryRouter,
   useRouteMatch,
+  useHistory,
```
Comment (Dmytro-Melnyshyn) [inline src/components/SearchModeNavigation/SearchModeNavigation.test.js:5]: Please remove it, it is not used anywhere.

**Review by Dmytro-Melnyshyn — CHANGES_REQUESTED** [review]

File: src/providers/LastSearchTermsProvider.js [hunk 5]
```diff
@@ -16,28 +17,33 @@ const LastSearchTermsProvider = ({ children }) => {
     setItem(`${namespace}/browse.lastSearch`, null);
     setItem(`${namespace}/search.lastOffset`, null);
     setItem(`${namespace}/browse.lastOffset`, null);
+    setItem(`${namespace}/search.segment`, null);
```
[the remark this answers is not in the document]
Comment (Dmytro-Melnyshyn) [reply src/providers/LastSearchTermsProvider.js:20]: It is also necessary to reset all search terms for each segment in `resetStorageAfterLogout` to not have them after login.

File: src/components/SearchModeNavigation/SearchModeNavigation.js [hunk 6]
```diff
@@ -16,38 +17,45 @@ import {
...
 
+  const onClick = useCallback((segment) => {
+    const isCurrentSegment = pathname === searchModeRoutesMap[segment];
+
+    if (onSearchModeSwitch) {
+      onSearchModeSwitch();
+    }
+
+    history.push({
+      pathname: searchModeRoutesMap[segment],
```
[the remark this answers is not in the document]
Comment (Dmytro-Melnyshyn) [reply src/components/SearchModeNavigation/SearchModeNavigation.js:37]: Please look at the scenario described above.

**Review by Dmytro-Melnyshyn — CHANGES_REQUESTED** [review]

**Review by Dmytro-Melnyshyn — APPROVED** [review]

**Pull request merged**